orderedmap.go

```diff
@@ -3,19 +3,19 @@ package orderedmap
 import (
```

Comment:
There are still unchecked `json.Unmarshal` calls such as on [this line](https://github.com/Pexeso/orderedmap/blob/00f886ea4c249aff10285c690e6ba4e73ccb2c8c/orderedmap.go#L177). Should these also be updated to be absolutely sure that no edge cases are missed?

Reply:
A lot of the `json.Unmarshal` calls are used to just check for valid JSON syntax, and their output values aren't even used, so I left those around. I could get rid of all the `json.Unmarshal` and the `setNumber` option completely and just always have the effects of `setNumber == true`. Would you prefer that?

Reply:
Ah that's fine then. LGTM

Reply:
Done.